### PR TITLE
WIP: Update README.md

### PR DESCRIPTION
**Pull request by AngyDev**
Related issue: #72 

I tried to add some documentation, I hope it's not a mess. It's not easy.
The two classes PointsHelper and HalfEdgeHelper are not documented, I'm not sure about how to explain them.

Feel free to cancel the PR if the documentation is not consistent. I understand 😅

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -118,6 +145,62 @@ TODO
...
 -->
 
+## GridMaterial
+
+_extends THREE.MeshPhongMaterial_
+
+A material with the same interface as `THREE.MeshPhongMaterial`. It's used to add a grid on the mesh, created by the method `csgGridShaderMixin` passing a shader. Default is enabled but if you want you can disable the grid with
+```js
+gridMat.enableGrid = false;
+```
```
Comment (gkjohnson) [inline README.md:155]: Here we can add a member section for `enableGrid` that describes its use like so:

````md
### .enableGrid

```js
enableGrid = false : Boolean
```

TODO description
````

File: README.md [hunk 2]
```diff
@@ -118,6 +145,62 @@ TODO
...
+## EdgesHelper
+
+_extends THREE.LineSegments_
+
+Helper class for generating a line to display the intersection edges. If no edges are provided then an empty array will be passed.
+
+### .setEdges
+
+```js
+setEdges( edges: EdgesGeometry[] ) : void
```
Comment (gkjohnson) [inline README.md:183]: This function should take an array of `Line3` objects, instead.

File: README.md [hunk 3]
```diff
@@ -118,6 +145,62 @@ TODO
...
+```
+
+Sets the geometry of the helper with the points of the intersection of the edges. First gets the points of the edges, disposes of geometry and then sets the attributes for this geometry from an array of points of the edges.
```
Comment (gkjohnson) [inline README.md:186]: I think we can keep this simple and just say that this function sets the list of lines to be visualized.

File: README.md [hunk 4]
```diff
@@ -68,14 +68,22 @@ An object with the same interface as `THREE.Mesh` but used to evaluate CSG opera
 > 
 > It is recommended to remove groups from a geometry before creating a brush if multi-material support is not required.
 
-<!--
 ## Operation
 
 _extends Brush_
 
-TODO
+This is an extension of the `Brush` class. With this class is possible to create a mesh and pass the CSG Operation constant to define which operation computes on the mesh. If you don't choose an operation, the default is `ADDITION`. The result is a Mesh whose effect is defined from the operation selected. You can see how it works in the _hierarchy_ example.
```
Comment (gkjohnson) [inline README.md:75]: We'll want to document the local member of `.operation`, as well:

````md
### .operation

```js
operation = ADDITION : CSGOperation
```
````

We'll also want to document the "insertBefore" and "insertAfter" functions.

File: README.md [hunk 5]
```diff
@@ -68,14 +68,22 @@ An object with the same interface as `THREE.Mesh` but used to evaluate CSG opera
...
 
--->
+```js
+const root = new Operation( new THREE.BoxBufferGeometry(10, 5, 0.5), new GridMaterial() );
+```
```
Comment (gkjohnson) [inline README.md:79]: I don't think this is needed - the implicit constructor of `Brush` should be enough, I think!

**Review by gkjohnson — COMMENTED** [review]
This is great, thank you!

> Feel free to cancel the PR if the documentation is not consistent. I understand 😅

This is all appreciated! It's really great to get all this stuff filled in and we can always improve things over time once we get feedback from other users on documentation

> The two classes `PointsHelper` and `HalfEdgeHelper` are not documented, I'm not sure about how to explain them.

The `PointsHelper` is similar to the other two documented helpers in that you will pass an array of Vector3s to `setPoints` and all those points will be visualized as spheres.

The "HalfEdgeHelper" takes the `HalfEdgeMap` object and visualizes the conectivity between triangles. It's probably okay to skip documenting this one for now.

File: README.md [hunk 6]
```diff
@@ -100,11 +108,30 @@ evaluate(
...
+The method gets as parameters a root, an Operation mesh and a target if it is provided, otherwise, a new `Brush` will be created. 
+First sets the `updateMatrixWolrd` of the root to true then calls the traverse function with the root parameter to `evaluate` the mesh and its children. 
+
+```js
+evaluate( brush, child, child.operation );
+```
+
+In this case, the arguments passed to `evaluate` is the `root` as `brushA`, the child as `brushB` and the `child.operation` as the operation to apply to the mesh.
+
+## OperationDebugData
```
Comment (gkjohnson) [inline README.md:130]: Lets at least document `enabled` and `intersectionEdges` for this class. I can fill in some of the content if it's not clear what it does.

**Comment by AngyDev** [comment]
Added your suggestions. I think it's a start. I also corrected the types of the helper methods.

**Comment by gkjohnson** [comment]
This is excellent and a huge help - thank you again!

**Pull request merged**